### PR TITLE
Feature/50 test framework

### PR DESCRIPTION
**Pull request by MadRabbit**
So, this pretty chunky thing closes several issues

1. #50 replaces tape with mocha
2. #63 changes all assertions to use `must` consistently with the `should` syntax
3. #41 makes all test files automatically have the `_test` suffix
4. #45 + #2 renames the `tests/` -> `test/`
5. adds the `test/mocha.opts` file as a standard way to configure mocha within projects
6. makes tests awesome again

### CODE REVIEW

File: base/test/middleware/jwt_test.mustache [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const fetch = require('node-fetch');
+const jwt   = require('jsonwebtoken');
```
Comment (nwinch) [inline base/test/middleware/jwt_test.mustache]: Do we need to align assignments?  This is probably one of "those" conversations, but we haven't done that on any other projects so to be safe I think let's not introduce the style here, but let's discuss it.

Comment (MadRabbit) [inline base/test/middleware/jwt_test.mustache]: Ah, sorry my bad. I normally align things in my projects, must have done this accidentally here. Will put this back

File: base/test/mocha.opts.mustache [hunk 2]
```diff
@@ -0,0 +1,3 @@
+--recursive
+--require test/helper
+--ui bdd
```
Comment (nwinch) [inline base/test/mocha.opts.mustache]: Mocha defaults to "bdd" so can probably remove this.

**Comment by nwinch** [comment]
@MadRabbit Review done with just a couple of minor things.  Lovely stuff.  LGTM 👍

**Pull request merged**